amqp/src/main/scala/akka/stream/alpakka/amqp/AmqpConnectorSettings.scala

```diff
@@ -4,39 +4,45 @@
 
 package akka.stream.alpakka.amqp
 
+import akka.annotation.InternalApi
+
+import scala.collection.immutable
+
 /**
  * Internal API
  */
+@InternalApi
 sealed trait AmqpConnectorSettings {
   def connectionProvider: AmqpConnectionProvider
-  def declarations: Seq[Declaration]
+  def declarations: immutable.Seq[Declaration]
 }
 
 sealed trait AmqpSourceSettings extends AmqpConnectorSettings
 
 final case class NamedQueueSourceSettings(
     connectionProvider: AmqpConnectionProvider,
     queue: String,
-    declarations: Seq[Declaration] = Seq.empty,
+    declarations: immutable.Seq[Declaration] = Nil,
```

Comment:
This is not ideal. I think instead of importing `s.c.immutable` we should add type and companion object alias in the connector base package object: 

```scala
package object amqp {
  type Seq[A] = immutable.Seq[A]
  val Seq = immutable.Seq
}
```

in `... /alpakka/ampq/package.scala` file.